bsip-0038.md

```diff
@@ -78,7 +88,121 @@ max_amount_to_sell = (debt * target_CR - collateral * feed_price)
                      / (target_CR * match_price - feed_price)
 ```
 
-The result is a rational number, need to be rounded up to an integer.
+The result is a rational number.
+
+Then, the maximum debt it wants to cover can be calculated as:
+
+```
+max_debt_to_cover = max_amount_to_sell * match_price
+```
+
+The result is a rational number as well.
+
+## Rounding on Maximums Calculation
+
+As described in [BSIP 35](bsip-0035.md), at last we need to convert the
+rational numbers to integers, so rounding is involved.
+
+When calculating maximum debt to cover, the goal is to go over the specified
+target but not go too far beyond.
+That said, if a short position got matched with a big limit order, after
+partially filled, its collateral ratio should be **just** higher than specified
+target collateral ratio. Which means if `max_debt_to_cover` has no
+fraction, need to plus it by one Satoshi; otherwise, need to round it up.
+An effectively same approach is to round down then add one Satoshi onto the
+result.
+
+With `max_debt_to_cover` in integer, `max_amount_to_sell` in integer should be
+calculated as:
+
+```
+max_amount_to_sell = round_up(max_debt_to_cover / match_price)
+```
+
+Then adjust `max_debt_to_cover` to be more accurate with:
+
+```
+max_debt_to_cover = round_down(max_amount_to_sell * match_price)
+```
+
+## Rounding on Order Matching, and Edge Cases
+
+Rounding rules about order matching are defined in [BSIP 35](bsip-0035.md).
+
+Generally, when two orders got matched, the order matching engine will favor
+the larger order while rounding.
+
+When a call order got matched with a limit order, if the call order has no
+`target_CR` option set, or has `target_CR` option set but covering debt is
+smaller than maximum debt to cover, both means the call order is larger,
+so its paid collateral will be rounded down, so its collateral ratio will
+increase after partially filled.
```

Comment:
I do not understand the conclusion "both means the call order is larger" for four reasons:

1. I do not see how "If the call order has no "target_CR option set" indicates anything about the size of the call order
2. if the "covering debt is smaller than the maximum debt to cover" it makes me think that this call order will only get partially filled. If that is correct, it makes me think that the call order is smaller than the matching limit order.
3. "so its paid collateral will be rounded down" matches up with the suggested approach to handle partial fills in BSIP-35
4. "so its collateral ratio will increase after partially filled." explicitly says that the call order will be partially filled.

If I have understood this BSIP correctly, partial filling of a call order means that the matching order is smaller than the call order.

Reply:
The first point makes sense. Updated. Thanks. Please check again.

Reply:
It makes sense to me now, thanks